src/utils/components/PolicyForm/PolicyWizard/components/InterfaceDetails.tsx

```diff
@@ -0,0 +1,156 @@
+  Popover,
+  TextInput,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+import { InterfaceType, NodeNetworkConfigurationInterface } from '@types';
+
+import BondConfiguration from '../../components/BondConfiguration';
+import IpConfiguration from '../../components/IpConfiguration';
+import NetworkState from '../../components/NetworkState';
+import PortConfiguration from '../../components/PortConfiguration';
+import { onInterfaceChangeType } from '../../constants';
+import { validateInterfaceName } from '../../utils';
+
+import BridgeType from './BridgeType';
+
+type InterfaceDetailsProps = {
+  id: number;
+  isInterfaceCreated?: boolean;
+  policyInterface?: NodeNetworkConfigurationInterface;
+  onInterfaceChange?: (updateInterface: onInterfaceChangeType) => void;
+  label?: string;
+};
+
+const InterfaceDetails: FC<InterfaceDetailsProps> = ({
+  id,
+  policyInterface,
+  onInterfaceChange,
+  isInterfaceCreated = true,
+  label,
+}) => {
+  const isBridgeType = [InterfaceType.LINUX_BRIDGE, InterfaceType.OVS_BRIDGE].includes(
+    policyInterface?.type,
+  );
+
+  const { t } = useNMStateTranslation();
+
+  const handleNameChange = (newName: string) => {
+    onInterfaceChange((draftInterface) => (draftInterface.name = newName));
+  };
+
+  const onSTPChange = (checked: boolean) => {
+    onInterfaceChange((draftInterface) => {
+      ensurePath(draftInterface, 'bridge.options');
+
+      draftInterface.bridge = {
+        options: {
+          stp: { enabled: checked },
+        },
+      };
+    });
+  };
+
+  const nameError = validateInterfaceName(policyInterface?.name);
+
+  return (
+    <>
+      {isBridgeType && (
+        <BridgeType
+          id={id}
+          policyInterface={policyInterface}
+          onInterfaceChange={onInterfaceChange}
+        />
+      )}
+      <FormGroup
+        label={label || t('Interface name')}
+        isRequired
+        fieldId={`policy-interface-name-${id}`}
+      >
+        <TextInput
+          isRequired
+          id={`policy-interface-name-${id}`}
+          name={`policy-interface-name-${id}`}
+          value={policyInterface?.name}
+          onChange={(_, newValue) => handleNameChange(newValue)}
+          isDisabled={isInterfaceCreated}
+        />
+        {nameError && (
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem icon={<RedExclamationCircleIcon />} variant={ValidatedOptions.error}>
+                {nameError}
+              </HelperTextItem>
+            </HelperText>
+          </FormHelperText>
+        )}
+      </FormGroup>
+      <NetworkState
+        id={id}
+        policyInterface={policyInterface}
+        onInterfaceChange={onInterfaceChange}
+      />
+
+      <IpConfiguration
+        id={id}
+        policyInterface={policyInterface}
+        onInterfaceChange={onInterfaceChange}
+      />
+
+      {policyInterface.type !== InterfaceType.ETHERNET && (
```

Comment:
You use `policyInterface.type` several times in this file. Consider storing it in a variable.